scripts/component_readme/readme_template.md

```diff
@@ -1,31 +1,45 @@
 # {{ name }}
 
-### Description
+## Description
 {{ description }}
 
-### Inputs / outputs
+## Inputs / outputs
 
+### Consumes
 {% if consumes %}
 **This component consumes:**
 
 {% for field_name, field in consumes.items() %}
 - {{ field.name }}: {{ field.type.value }}
 {% endfor %}
 {% else %}
-**This component consumes no data.**
+_**This component does not consume specific data.**_
 {% endif %}
 
+{% if is_consumes_generic %}
+**This component consumes generic data**
+- <field_name>: <mapped_field_name>
+{% endif %}
+
+
+### Produces
+
 {% if produces %}
 **This component produces:**
 
 {% for field_name, field in produces.items() %}
 - {{ field.name }}: {{ field.type.value }}
 {% endfor %}
 {% else %}
-**This component produces no data.**
+_**This component does not produce specific data.**_
 {% endif %}
 
-### Arguments
+{% if is_produces_generic %}
+**This component produces generic data**
+- <field_name>: <field_schema>
```

Comment:
The produce implementation can use both, a column mapping and a column/data type mapping. What you can use depends on the component implementation itself. Can we add a description for column name, data type mapping too?

Reply:
good call! added a description based on the docs